proxy/README.md

```diff
@@ -0,0 +1,99 @@
+# `cc-proxy`
+
+`cc-proxy` is a daemon offering access to the VM agent to multiple clients on
+the host. It exposes an API accessible through an `AF_UNIX` socket.
+
+This socket can be found at: `${localstatesdir}/run/cc-oci-runtime/proxy.sock`.
+
+## Protocol
+
+The proxy protocol is composed of JSON messages: requests and responses. Each
+of these message is on a separate line. The snippet below is showing two
+requests (`hello` and `hyper` commands) and their corresponding responses:
+
+```
+{ "id": 0, "command": "hello", "data": { "containerId": "foo", "ctlSerial": "/tmp/sh.hyper.channel.0.sock", "ioSerial": "/tmp/sh.hyper.channel.1.sock"  } }
+{"id":0,"success":true}
+{ "id": 1, "command": "hyper", "data": { "name": "ping" }}
+{"id":1,"success":true}
+```
+
+Requests have 3 fields: `id`, `command` and `data`.
+
+```
+type Request struct {
+	Id      uint             `json:"id"`
+	Command string           `json:"command"`
+	Data    *json.RawMessage `json:"data"`
+}
+```
+
+The `id` identifies the request and the corresponding response will have the
+same `id`. Each request carries a `command` which can have `data` associated,
+not unlike what a simple JSON RPC protocol would have.
+
+Responses have 4 fields: `id`, `success`, `error` and `data`
+
+```
+type Response struct {
+	Id      uint                   `json:"id"`
+	Success bool                   `json:"success"`
+	Error   string                 `json:"error,omitempty"`
+	Data    map[string]interface{} `json:"data,omitempty"`
+}
+```
+
+Unsurprisingly, the response has the result of a command, with `success`
+indicating if the request has succeeded for not. If `success` is `true`, the
+response can carry additional return values in `data`. If success if `false`,
+`error` may contain an error string.
+
+### `hello`
+
+```
+type helloCmd struct {
+        ContainerId string `json:containerId`
+        CtlSerial   string `json:ctlSerial`
+        IoSerial    string `json:ioSerial`
+}
+```
+
+The `hello` command is issued first after connecting to the proxy socket.  It's
```

Comment:
So it's going to sound like nitpicking, but to me `hello` is not a command but a command payload. The command is the request and it contains a command/ID string.